[🐛 Bug]: [py] Mypy error - WebDriverWait.driver has attribute

### Description


The following error from mypy occurs for `WebDriverWait.__repr__`

**error: Item "WebElement" of the upper bound "WebDriver | WebElement" of type variable "D" has no attribute "session_id"  [union-attr]**

Explanation:
If you pass a WebElement into `WebDriverWait`, the repr will not be able to properly access the `session_id` attribute.  Mypy is detecting this.
```python
    def __repr__(self):
        return f'<{type(self).__module__}.{type(self).__name__} (session="{self._driver.session_id}")>'
```

My guess is there likely needs to be additional logic to access `driver` from `WebElement.parent` in those cases, or `session_id` needs to be exposed as an attribute of `WebElement`.

### Reproducible Code

```shell
def __repr__(self):
        return f'<{type(self).__module__}.{type(self).__name__} (session="{self._driver.session_id}")>'
```

--- a/py/selenium/webdriver/remote/webelement.py
+++ b/py/selenium/webdriver/remote/webelement.py
@@ -77,7 +77,11 @@ class WebElement(BaseWebElement):
         self._id = id_
 
     def __repr__(self):
-        return f'<{type(self).__module__}.{type(self).__name__} (session="{self._parent.session_id}", element="{self._id}")>'
+        return f'<{type(self).__module__}.{type(self).__name__} (session="{self.session_id}", element="{self._id}")>'
+
+    @property
+    def session_id(self) -> str:
+        return self._parent.session_id
 
     @property
     def tag_name(self) -> str:

--- a/py/selenium/webdriver/support/wait.py
+++ b/py/selenium/webdriver/support/wait.py
@@ -81,7 +81,7 @@ class WebDriverWait(Generic[D]):
         # avoid the divide by zero
         if self._poll == 0:
             self._poll = POLL_FREQUENCY
-        exceptions = list(IGNORED_EXCEPTIONS)
+        exceptions: list = list(IGNORED_EXCEPTIONS)
         if ignored_exceptions:
             try:
                 exceptions.extend(iter(ignored_exceptions))
@@ -89,7 +89,7 @@ class WebDriverWait(Generic[D]):
                 exceptions.append(ignored_exceptions)
         self._ignored_exceptions = tuple(exceptions)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f'<{type(self).__module__}.{type(self).__name__} (session="{self._driver.session_id}")>'
 
     def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = "") -> T:

